progress.go

```diff
@@ -92,9 +95,17 @@ func (p *Progress) Report(s Stat) {
 	p.curM.Lock()
 	p.cur.Add(s)
 	cur := p.cur
+	needUpdate := false
+	if time.Since(p.lastUpdate) > minTickerTime {
```

Comment:
also right now this code that you are adding is mutexed by `curM`. Is that intentional/necessary? I don't see why.